onnx/defs/data_propagators.h

```diff
@@ -0,0 +1,29 @@
+#include "onnx/defs/shape_inference.h"
+
+namespace ONNX_NAMESPACE {
+
+inline void PropagateShapeDataFromInputToOutput(DataPropagationContext& ctx, int idx) {
+  // propogate input data
+  const auto input_data = ctx.getGeneratedShapeData(idx);
+  if (input_data != nullptr) {
+    TensorShapeProto tp;
+    tp.CopyFrom(*input_data);
+    ctx.addGeneratedShapeData(0, std::move(tp));
+  }
+}
+
+inline void ShapeDataPropagator(DataPropagationContext& ctx) {
```

Comment:
what is the purpose of this function?

Reply:
after reading the rest of the PR now I understand this is the data propagator for shape op... Please update the function name to make it more clear and add some comments